include/zephyr/bluetooth/att.h

```diff
@@ -101,6 +101,20 @@ extern "C" {
 #define BT_ATT_LAST_ATTRIBUTE_HANDLE            0xffff
 #define BT_ATT_LAST_ATTTRIBUTE_HANDLE __DEPRECATED_MACRO BT_ATT_LAST_ATTRIBUTE_HANDLE
 
+/** Converts a ATT error to string.
+ *
+ * The error codes are described in the Bluetooth Core specification,
+ * Vol 3, Part F, Section 3.4.1.1.
+ *
+ * The ATT and GATT documentation found in Vol 4, Part F and
+ * Part G describe when the different error codes are used.
+ *
+ * See also the defined BT_ATT_ERR_* macros.
+ *
+ * @return The string representation of the ATT error code.
+ */
+const char *bt_att_err_to_str(uint8_t att_err);
```

Comment:
Does it make sense to also have `bt_gatt_err_to_str` which basically just calls this with `-att_err`?

Reply:
Yeah, that sounds like a good idea. That may help some users not knowing the difference between ATT/GATT

Reply:
Indeed, and most users at working at the GATT level. 

The GATT function may want to assume that users are unaware of what they are doing do be something like 
```c
const char *bt_gatt_err_to_str(uint8_t gatt_err)
{
    bt_att_err_to_str(gatt_err < 0 ? -gatt_err : gatt_err)
}
```

so that it's works for both e.g. `-2` and `2` for `BT_ATT_ERR_READ_NOT_PERMITTED`

Reply:
The biggest challenge there is of course having users know the difference between GATT errors and errno values...